querydsl-core/src/main/java/com/mysema/query/group/GroupBy.java

```diff
@@ -106,43 +103,200 @@
     }
 
     /**
-     * Create a new aggregating set expression
+     * Create a new aggregating set expression using a backing LinkedHashMap
      *
      * @param expression
      * @return
      */
     public static <E> AbstractGroupExpression<E, Set<E>> set(Expression<E> expression) {
-        return new GSet<E>(expression);
+        return GSet.createLinked(expression);
     }
 
     public static <E, F> GroupExpression<E, Set<F>> set(GroupExpression<E, F> groupExpression) {
-        return new MixinGroupExpression<E, F, Set<F>>(groupExpression, new GSet<F>(groupExpression));
+        return new MixinGroupExpression<E, F, Set<F>>(groupExpression, GSet.createLinked(groupExpression));
     }
 
     /**
-     * Create a new aggregating map expression
+     * Create a new aggregating set expression using a backing TreeSet
+     *
+     * @param expression
+     * @return
+     */
+    public static <E extends Comparable<? super E>> AbstractGroupExpression<E, SortedSet<E>> sortedSet(Expression<E> expression) {
+        return GSet.createSorted(expression);
+    }
+
+    /**
+     * Create a new aggregating set expression using a backing TreeSet
+     *
+     * @param groupExpression
+     * @return
+     */
+    public static <E, F extends Comparable<? super F>> GroupExpression<E, SortedSet<F>> sortedSet(GroupExpression<E, F> groupExpression) {
+        return new MixinGroupExpression<E, F, SortedSet<F>>(groupExpression, GSet.createSorted(groupExpression));
+    }
+
+
+    /**
+     * Create a new aggregating set expression using a backing TreeSet using the given comparator
+     *
+     * @param expression
+     * @param comparator
+     * @return
+     */
+    public static <E> AbstractGroupExpression<E, SortedSet<E>> sortedSet(Expression<E> expression, Comparator<? super E> comparator) {
+        return GSet.createSorted(expression, comparator);
+    }
+
+    /**
+     * Create a new aggregating set expression using a backing TreeSet using the given comparator
+     *
+     * @param groupExpression
+     * @param comparator
+     * @return
+     */
+    public static <E, F> GroupExpression<E, SortedSet<F>> sortedSet(GroupExpression<E, F> groupExpression, Comparator<? super F> comparator) {
+        return new MixinGroupExpression<E, F, SortedSet<F>>(groupExpression, GSet.createSorted(groupExpression, comparator));
+    }
+
+
+    /**
+     * Create a new aggregating map expression using a backing LinkedHashMap
      *
      * @param key
      * @param value
      * @return
      */
-    @WithBridgeMethods(value=Expression.class,castRequired=true)
```

Comment:
I will fix that
On 4 Mar 2015 01:17, "Ruben Dijkstra" <notifications@github.com> wrote:

> In querydsl-core/src/main/java/com/mysema/query/group/GroupBy.java
> <https://github.com/querydsl/querydsl/pull/1232#discussion_r25738075>:
>
> >       *
> >       * @param key
> >       * @param value
> >       * @return
> >       */
> > -    @WithBridgeMethods(value=Expression.class,castRequired=true)
>
> Are you sure the bridge method is not necessary?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/querydsl/querydsl/pull/1232/files#r25738075>.
>